src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -189,7 +189,10 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]) -> None:
         addr = dd.ip if dd.ip else self._inventory_get_fqdn(dd.hostname)
         port = dd.ports[0] if dd.ports else self.DEFAULT_SERVICE_PORT
         spec = cast(GrafanaSpec, self.mgr.spec_store[dd.service_name()].spec)
+        adming_gw_cnt = len(self.mgr.cache.get_daemons_by_service('admin-gateway'))
+        url_prefix = '/grafana' if adming_gw_cnt > 0 else ''
         service_url = build_url(scheme=spec.protocol, host=addr, port=port)
+        service_url = f'{service_url}{url_prefix}'
```

Comment:
this parameter in fact is called prefix `--web.route-prefix` (the same terminology is used in our dashboard and the rest of the monitoring stack). It's called prefix because it acts as a prefix for all the URLs so instead of visiting "/" you use `/prefix/<sub-url>` instead.